### PR TITLE
[v5] Call close even if the channel never opened

### PR DESCRIPTION
**Pull request by masad-frost**
Why
===
Channels should get a signal that they're closed regardless of whether they ever opened or not. Getting a channel is optimistic and you can start sending requests on it and use it, so we need to do proper cleanup (i.e. rejecting all requests) when the client closes before the channel actually opens.

What changed
============
- Channels that are inflight get added to a set of `this.openningChannels` while their open requests are inflight.
- When the we close the connection we call the close handler on each of the unoppened channels

Test plan
=========
run this code
```
const c = new Crosis.Client();

c.connect({
  token: TOKEN_HERE,
}).then(() => {
  console.log('connected');
  const chan = c.openChannel({ name: 'exec' });
  chan.on('open', () => {
    console.log('channel opened');
  });
  chan.on('close', () => {
    console.log('chan close called');
  });

  c.ws.close();
});
```
Notice that `'chan close called'` is logged without `'channel opened'` being called

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -438,6 +440,16 @@ export class Client extends EventEmitter {
...
     });
 
+    // These are channels that never opened, but we should emit the
+    // close event on them anyway to do proper cleanup
+    this.inflightChannels.forEach((chan) => {
+      chan.onClose({
+        initiator: 'client',
+        clientCloseReason: closeResult.closeReason,
+      });
+      this.inflightChannels.delete(chan);
```
Comment (masad-frost) [inline src/client.ts:450]: Not sure if we should delete them here since the client can't be re-used anyway

Comment (masad-frost) [reply src/client.ts:450]: We just lose some information on the dead client but it's fine

**Comment by moudy** [comment]
is this is the equivalent of the v6 api being called with an error on initial open?

**Comment by masad-frost** [comment]
Yeah pretty much!

File: src/client.ts (hunk 1 above)
Comment (moudy) [reply src/client.ts:450]: seems fine

**Pull request merged**